src/ui/ui_mappings.js

```diff
@@ -0,0 +1,37 @@
+import _ from 'lodash';
+
+class MappingsCollection {
+  constructor() {
+    this._defaultMappings = {
+      '_default_': {
+        'dynamic': 'strict'
+      },
+      config: {
+        dynamic: true,
+        properties: {
+          buildNum: {
+            type: 'keyword'
+          }
+        }
+      },
+    };
+    this._currentMappings = _.cloneDeep(this._defaultMappings);
+  }
+
+  getCombined = () => {
+    return this._currentMappings;
+  }
+
+  register = (newMappings, options = {}) => {
+    Object.keys(this._currentMappings).forEach(key => {
+      if (newMappings.hasOwnProperty(key)) {
+        const pluginPartial = options.plugin ? `registered by plugin ${options.plugin} ` : '';
+        throw new Error(`Mappings for ${key} ${pluginPartial}have already been defined`);
+        return;
+      }
+    });
+    Object.assign(this._currentMappings, newMappings);
+  }
+}
+
+export { MappingsCollection };
```

Comment:
In the future, we should not export raw objects like this.  This is not valid JavaScript, it's just something that babel supports because it happens to behave similarly to CommonJS modules.  Export the class itself instead.